sbin/submitty_autograding_shipper.py

```diff
@@ -789,6 +805,317 @@ def shipper_process(my_name,my_data,full_address,which_untrusted):
+
+    try:
+        autograding_utils.just_write_grade_history(history_json_path,
+                                                   h['gradeable_deadline'],
+                                                   h['submission'],
+                                                   h['seconds_late'],
+                                                   h['first_access'],
+                                                   h['access_duration'],
+                                                   queue_obj['queue_time'],
+                                                   'BATCH' if queue_obj.get('regrade', False) else 'INTERACTIVE',
+                                                   dateutils.write_submitty_date(grading_began),
+                                                   int(queue_obj['waittime']),
+                                                   dateutils.write_submitty_date(grading_finished),
+                                                   int(grading_time),
+                                                   autograde_result_msg,
+                                                   queue_obj.get('revision', None))
+
+        results_zip_path = os.path.join(base_dir, 'results.zip')
+        autograding_utils.zip_my_directory(results_dir, results_zip_path)
+        packer_unpacker.unpack_grading_results_zip('(short-circuit)',
+                                                   '(short-circuit)',
+                                                   results_zip_path)
+    except Exception as e:
+        autograding_utils.log_message(AUTOGRADING_LOG_PATH,
+                                      message=f"Short-circuit failed for {gradeable_id} (check stack traces). Falling back to standard grade.",
+                                      job_id=job_id)
+        autograding_utils.log_stack_trace(AUTOGRADING_STACKTRACE_PATH,
+                                          trace=traceback.format_exc(),
+                                          job_id=job_id)
+        return False
+    finally:
+        shutil.rmtree(base_dir, ignore_errors=True)
+
+    autograding_utils.log_message(AUTOGRADING_LOG_PATH,
+                                  message=f"Successfully short-circuited {gradeable_id}!",
+                                  job_id=job_id)
+    return True
+
+
+def write_grading_outputs(testcases: list,
+                          testcase_outputs: list,
+                          results_json: str,
+                          grade_txt: str) -> str:
+    """Write the grading output data to the specified paths."""
+
+    ###########################################################################
+    #
+    # NOTE: Editing this file? Make sure that this function stays in-sync with
+    #       grading/main_validator.cpp::validateTestCase()
+    #
+    ###########################################################################
+
+    results = {'testcases': testcase_outputs}
+    with open(results_json, 'w') as fd:
+        json.dump(results, fd)
```

Comment:
We usually write json files with `indent=4` for ease of debugging (probably not best practice, though).

Reply:
@hdrodz97 are you making this quick change?

Reply:
Just put it in